Wheels: ADIOS2 v2.10.1

ax3l opened the pull request:
Build wheels with ADIOS2 v2.9.2+, which fixes c-blosc2 issues with null-blocks.
Build with 2.10.0, which fixes static builds with ADIOS2 & c-blosc2 on Windows.

Note: since we do not bump the version of the already released 0.15.2 release, this will not create new wheels. Once we publish 0.15.3, this will be included.

Note: ADIOS2 BP5 supports will be available for Windows with the ADIOS 2.10+ release series.

ax3l committed:
Wheels: ADIOS2 v2.9.2

Build wheels with ADIOS2 v2.9.2, which fixes c-blosc2 issues with
null-blocks.

ax3l commented:
Probably something is different on this platform with respect to fundamental char type defaults:
https://pybind11.readthedocs.io/en/stable/advanced/cast/strings.html#passing-bytes-to-c

and we might lack an overload or so.

ax3l committed:
C-Blosc2 v2.11.1

ax3l committed:
ADIOS2: Remove installed FindBlosc2.cmake Again

ax3l committed:
[Patch] C-Blosc2 Static

ax3l committed:
No Patch: Static C-Blosc2

ax3l commented:
@vicentebolea can you help me with this all-static deployment of ADIOS 2.9.2? It is related to the upstream issues I opened :)

ax3l committed:
Windows: More Blosc2 Updates

ax3l commented:
Does BP5 not work on Windows for ADIOS2 2.9.2?
```cmake
# BP5
if(ADIOS2_USE_BP5 AND NOT WIN32)
  set(ADIOS2_HAVE_BP5 TRUE)
endif()
```

We want to switch our default output to BP5 after our last discussion w/ @pnorbert :) Serial read support would be awesome on any platform :)

Update: confirmed, will be part of the ADIOS2 v2.10 release series (already in ADIOS2 development branch).

ax3l commented on library_builders.bat:
```diff
@@ -51,6 +49,11 @@ exit /b 0
 
+  :: CMake Config package of C-Blosc 2.10.1+ only
+  :: https://github.com/ornladios/ADIOS2/issues/3903
+  rmdir /s /q "%BUILD_PREFIX%/ADIOS2/lib/cmake/adios2/FindBlosc2.cmake"
+  if errorlevel 1 exit 1
```
```suggestion
  :: rmdir /s /q "%BUILD_PREFIX%/ADIOS2/lib/cmake/adios2/FindBlosc2.cmake"
  :: if errorlevel 1 exit 1
```

ax3l committed:
Windows: keep FindBlosc2.cmake

ax3l committed:
Win: Try ADIOS2 v2.10.0-rc1

ax3l commented:
ADIOS 2.10.0-rc1 on Windows:
```
2024-01-30T18:43:37.2016284Z   adios2_c_variable.cpp
2024-01-30T18:43:37.4167988Z LINK : fatal error LNK1104: cannot open file 'ffi.lib' [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\bpls.vcxproj]
2024-01-30T18:43:37.4954992Z   main.cpp
2024-01-30T18:43:38.2227279Z   Reorganize.cpp
2024-01-30T18:43:38.2323129Z   adios2_c_attribute.cpp
2024-01-30T18:43:38.5659633Z   IO.cpp
2024-01-30T18:43:39.1746622Z   adios2_c_operator.cpp
2024-01-30T18:43:39.4554626Z   Utils.cpp
2024-01-30T18:43:39.6007589Z   Operator.cpp
2024-01-30T18:43:39.9410487Z   Generating Code...
2024-01-30T18:43:40.0518750Z   Generating Code...
2024-01-30T18:43:40.3069063Z LINK : fatal error LNK1104: cannot open file 'ffi.lib' [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\adios_reorganize.vcxproj]
```

https://github.com/ornladios/ADIOS2/issues/4019

ax3l committed:
ADIOS2 Windows: Disable libFFI

franzpoeschel commented:
I think I found the problem and it looks like a CMake bug. I was able to reproduce the same issue with CMake 3.23 on Windows (install Blosc2 and ADIOS2 statically --> try configuring openPMD-api with these dependencies). I checked the CMake scripts and they looked fine, so instead I upgraded to CMake 3.28, and that seems to work now.

ax3l commented:
Awesome!

Hm, CI installs CMake 3.28.1 already... Let me check if it picks up the right version.

ax3l committed:
Print CMake Version

vicentebolea commented:
@ax3l it appears that we were exporting blosc2 unnecessarily, and when doing out find_dependency defaulted to module as per cmake 3.24, rather than config. Solution seems to be not to export it anyways as our bloscs2 dep is private.

ax3l commented:
Thank you, @vicentebolea!

Is there potentially a patch I can apply for this?
Note that I am doing an **all-static build** for pip here, so to some extent I think we need to export blosc2 at least on the link lines.

vicentebolea commented:
@ax3l we are release adios 2.10 (final release) very soon. Otherwise I think that those changes are patchable (in 2.10 and 2.9.2)

ax3l commented:
@vicentebolea Awesome - can I test a branch / RC here? :)

ax3l commented:
@vicentebolea Can I test a branch / RC for ADIOS2 2.10 already? :)

ax3l committed:
ADIOS2: `release_210` Branch

vicentebolea commented:
> @vicentebolea Can I test a branch / RC for ADIOS2 2.10 already? :)

Current release_210 should have the fix for the windows static build

ax3l commented:
@vicentebolea thanks, I pushed to use `release_210` and still see in an all-static build:
- c-blosc2 static
- adios2 static
- openPMD-api static
- openPMD-api Python module: shared

the same issue.

Snippet from the Windows CI in this PR:
```
2024-03-12T18:13:01.6247052Z   CMake Error at C:/Program Files (x86)/ADIOS2/lib/cmake/adios2/adios2-targets.cmake:79 (set_target_properties):
2024-03-12T18:13:01.6248634Z     The link interface of target "adios2::core" contains:
2024-03-12T18:13:01.6251135Z 
2024-03-12T18:13:01.6253673Z       Blosc2::blosc2_static
2024-03-12T18:13:01.6255673Z 
2024-03-12T18:13:01.6258466Z     but the target was not found.  Possible reasons include:
2024-03-12T18:13:01.6260527Z 
2024-03-12T18:13:01.6263217Z       * There is a typo in the target name.
2024-03-12T18:13:01.6265554Z       * A find_package call is missing for an IMPORTED target.
2024-03-12T18:13:01.6267457Z       * An ALIAS target is missing.
2024-03-12T18:13:01.6269395Z 
2024-03-12T18:13:01.6271796Z   Call Stack (most recent call first):
2024-03-12T18:13:01.6274492Z     C:/Program Files (x86)/ADIOS2/lib/cmake/adios2/adios2-config-common.cmake:198 (include)
2024-03-12T18:13:01.6277493Z     C:/Program Files (x86)/ADIOS2/lib/cmake/adios2/adios2-config.cmake:29 (include)
2024-03-12T18:13:01.6278618Z     CMakeLists.txt:443 (find_package)
```

vicentebolea commented:
@ax3l can you try the 2.10.0-rc1 and when find_package(ADIOS2) add a CONFIG.

vicentebolea commented:
Try this branch for fixing the windows build: https://github.com/ornladios/ADIOS2/pull/4093

vicentebolea commented on library_builders.bat:
```diff
@@ -17,14 +17,15 @@ exit /b 0
-  curl -sLo adios2-2.9.1.zip ^
-    https://github.com/ax3l/ADIOS2/archive/refs/heads/release-2.9.1-bp-wheels.zip
-  powershell Expand-Archive adios2-2.9.1.zip -DestinationPath dep-adios2
+  curl -sLo adios2-2.10.0.zip ^
+    https://github.com/ornladios/ADIOS2/archive/refs/heads/release_210.zip
```
```suggestion
  curl -sLo adios2-2.10.0.zip ^
    https://github.com/vicentebolea/ADIOS2/archive/refs/heads/fix-static-blosc2-build.zip
```

ax3l committed:
Use Vincente's patch

Co-authored-by: Vicente Bolea <vicente.bolea@gmail.com>

ax3l commented on library_builders.bat:
```diff
@@ -17,14 +17,15 @@ exit /b 0
-  powershell Expand-Archive adios2-2.9.1.zip -DestinationPath dep-adios2
+  curl -sLo adios2-2.10.0.zip ^
+    https://github.com/vicentebolea/ADIOS2/archive/refs/heads/fix-static-blosc2-build.zip
+  powershell Expand-Archive adios2-2.10.0.zip -DestinationPath dep-adios2
```
```suggestion
  curl -sLo adios2-2.10.1.zip ^
    https://github.com/vicentebolea/ADIOS2/archive/refs/heads/fix-static-blosc2-build.zip
  powershell Expand-Archive adios2-2.10.1.zip -DestinationPath dep-adios2
```

ax3l commented on library_builders.sh:
```diff
@@ -103,10 +103,8 @@ function build_adios1 {
-    curl -sLo adios2-2.9.1.tar.gz \
-        https://github.com/ax3l/ADIOS2/archive/refs/heads/release-2.9.1-bp-wheels.tar.gz
+    curl -sLo adios2-2.9.2.tar.gz \
+        https://github.com/ornladios/ADIOS2/archive/v2.9.2.tar.gz
```
```suggestion
    curl -sLo adios2-2.10.1.tar.gz \
        https://github.com/ornladios/ADIOS2/archive/v2.10.1.tar.gz
```

ax3l committed:
ADIOS2: v2.10.1

vicentebolea commented on library_builders.bat:
```diff
@@ -17,14 +17,15 @@ exit /b 0
-    https://github.com/ax3l/ADIOS2/archive/refs/heads/release-2.9.1-bp-wheels.zip
-  powershell Expand-Archive adios2-2.9.1.zip -DestinationPath dep-adios2
+  curl -sLo adios2-2.10.1.zip ^
+    https://github.com/vicentebolea/ADIOS2/archive/refs/heads/fix-static-blosc2-build.zip
```
This needs to be updated

ax3l replied:
Done

ax3l committed:
Windows: Update ADIOS2 v2.10.1 URL

ax3l committed:
Update GH Actions

ax3l committed:
Python 3.8 -> 3.13

ax3l committed:
cibuildwheel: 2.16.2 -> 2.21.2

ax3l committed:
`-DADIOS2_USE_Campaign=OFF`

Issues:
- https://github.com/ornladios/ADIOS2/issues/4360
- https://github.com/ornladios/ADIOS2/issues/4148

ax3l committed:
Win: Compile Release

Install was already Release

ax3l committed:
Window: Fix unzipped ADIOS2 path

ax3l commented:
32bit windows compiles of ADIOS2 v2.10.1 still fail, with an error both in EVPATH and in BP5: https://github.com/ornladios/ADIOS2/issues/4105#issuecomment-2398929642

ax3l committed:
`actions/upload-artifact@v4` Names

ax3l committed:
`macos-11` -> `macos-12`
